src/code/z_play.c

```diff
@@ -1,81 +1,1575 @@
+                this->envCtx.screenFillColor[3] = 0;
+                this->transitionMode = TRANS_MODE_FILL_IN;
+            } else {
+                this->envCtx.screenFillColor[3] = 255;
+                this->transitionMode = TRANS_MODE_FILL_OUT;
+            }
+            break;
+
+        case TRANS_MODE_INSTANT:
+            if (this->transitionTrigger != TRANS_TRIGGER_END) {
+                STOP_GAMESTATE(&this->state);
+                SET_NEXT_GAMESTATE(&this->state, Play_Init, sizeof(PlayState));
+                gSaveContext.save.entrance = this->nextEntrance;
+                this->transitionTrigger = TRANS_TRIGGER_OFF;
+                this->transitionMode = TRANS_MODE_OFF;
+            } else {
+                gTrnsnUnkState = 0;
+                Game_SetFramerateDivisor(&this->state, 3);
+                this->transitionTrigger = TRANS_TRIGGER_OFF;
+                this->transitionMode = TRANS_MODE_OFF;
+            }
+            break;
+
+        case TRANS_MODE_INSTANCE_WAIT:
+            if (gSaveContext.cutsceneTransitionControl != 0) {
+                this->transitionMode = TRANS_MODE_INSTANCE_RUNNING;
+            }
+            break;
+
+        case TRANS_MODE_SANDSTORM_INIT:
+            if (this->transitionTrigger != TRANS_TRIGGER_END) {
+                this->envCtx.sandstormState = 1;
+                this->transitionMode = TRANS_MODE_SANDSTORM;
+            } else {
+                this->envCtx.sandstormState = 2;
+                this->envCtx.sandstormPrimA = 255;
+                this->envCtx.sandstormEnvA = 255;
+                this->transitionMode = TRANS_MODE_SANDSTORM;
+            }
+            break;
+
+        case TRANS_MODE_SANDSTORM:
+            func_8019F128(NA_SE_EV_SAND_STORM - SFX_FLAG);
+            if (this->transitionTrigger == TRANS_TRIGGER_END) {
+                if (this->envCtx.sandstormPrimA < 110) {
+                    gTrnsnUnkState = 0;
+                    Game_SetFramerateDivisor(&this->state, 3);
+                    this->transitionTrigger = TRANS_TRIGGER_OFF;
+                    this->transitionMode = TRANS_MODE_OFF;
+                }
+            } else {
+                if (this->envCtx.sandstormEnvA == 255) {
+                    STOP_GAMESTATE(&this->state);
+                    SET_NEXT_GAMESTATE(&this->state, Play_Init, sizeof(PlayState));
+                    gSaveContext.save.entrance = this->nextEntrance;
+                    this->transitionTrigger = TRANS_TRIGGER_OFF;
+                    this->transitionMode = TRANS_MODE_OFF;
+                }
+            }
+            break;
+
+        case TRANS_MODE_SANDSTORM_END_INIT:
+            if (this->transitionTrigger == TRANS_TRIGGER_END) {
+                this->envCtx.sandstormState = 4;
+                this->envCtx.sandstormPrimA = 255;
+                this->envCtx.sandstormEnvA = 255;
+                this->transitionMode = TRANS_MODE_SANDSTORM_END;
```

Comment:
```suggestion
                this->envCtx.sandstormEnvA = 255;
                // "It's here!!!!!!!!!"
                (void)"来た!!!!!!!!!!!!!!!!!!!!!";
                this->transitionMode = TRANS_MODE_SANDSTORM_END;
```
From OoT, we know this is where this string goes. Should we put it here like this?